### DISCUSSION TITLE
component discovery in subdirectories

### DISCUSSION THREAD

**Discussion by dgolda** [WeblateOrg/weblate#6754] [Q&A]
I have a project with gettext translation files like:
```
backend/aaa/aaa-api/src/main/resources/i18n/de/aaa-api.po
backend/aaa/aaa-api/src/main/resources/i18n/es/aaa-api.po
backend/aaa/aaa-api/src/main/resources/i18n/templates/aaa-api.pot

backend/bbb-rest-api/src/main/resources/i18n/de/bbb-rest-api.pot
backend/bbb-rest-api/src/main/resources/i18n/es/bbb-rest-api.pot
backend/bbb-rest-api/src/main/resources/i18n/templates/bbb-rest-api.pot

```
and so on

Plus special for first weblate component:
```
translations/po/de/!default.po
translations/po/templates/!default.pot
```

I've created projects and add manually component for !default.po and .pot. No problems.
But now I try to setup component discovery in this component and cannot manage.

Every example assume, that all components are in same parent directory, but for me I cannot specify such directory, so I have to used some pattern.

For regex and base file for new translations I tried (language filter always ^(de|es)$ ):
```
(.*)/src/main/resources/i18n/(?P<language>[^/.]*)/(?P<component>[^/]*)\.po
$1/src/main/resources/i18n/templates/{{component}}.pot
```
No components were matched!

```
src/main/resources/i18n/(?P<language>[^/.]*)/(?P<component>[^/]*)\.po
src/main/resources/i18n/templates/{{component}}.pot
```
No components were matched!

even
```
(?P<path>.*)/src/main/resources/i18n/(?P<language>[^/.]*)/(?P<component>[^/]*)\.po
{{path}}/src/main/resources/i18n/templates/{{component}}.pot
```
Failed to render template: Undefined variable: "path"

When I enter pattern with path to first files:
```
backend/aaa/aaa-api/src/main/resources/i18n/(?P<language>[^/.]*)/(?P<component>[^/]*)\.po
backend/aaa/aaa-api/src/main/resources/i18n/templates/{{component}}.pot
```
Then files are found - "Following components would be created"

Any suggestions?
Maybe it is a new feature?

**Comment by nijel** [comment]
Please upgrade, this was fixed in ba842cf5d97d32490e7fc399c0994b5c4416588d, see #6270 for more details.

**Comment by manuel-buildinglink** [comment]
We were trying to find all the translation files within our repo which could appear in different folder levels, all of them live in folders with this format `/lang/en.json`, so in order to accomplish that this is the regex we used:

`(?P<originalHierarchy>.+/)(?P<component>[^/]*)/lang/(?P<language>[^/.]*)\.json`

I Hope Weblate includes that more clearly in the documentation, it's a time saver @nijel

**Reply by nijel** [reply]
I've recently improved that in https://github.com/WeblateOrg/weblate/pull/9224, see https://docs.weblate.org/en/latest/admin/addons.html#component-discovery-examples. Further suggestions are welcome.
